python/setup.cfg

```diff
@@ -46,6 +46,4 @@ skip=
 
 [options]
 packages = find:
-install_requires =
-    numpy
```

Comment:
I think it should be possible to use KvikIO without CuPy and preferable also without NumPy.

Reply:
Gotcha. Though nvCOMP is still getting installed with this as well, right? If so, it seems we still need to list this here for nvCOMP.

Reply:
True. My idea is to only require numpy and cupy when the user imports `kvikio.nvcomp` explicitly. Notice, [`__init.__.py`](https://github.com/rapidsai/kvikio/blob/branch-22.10/python/kvikio/__init__.py) does not import nvcomp.

Reply:
Think if we are going to go this route, it would make sense to include them under [optional dependencies]( https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies ) (perhaps with the label `nvcomp`?).

Reply:
That could make sense if it doesn't get too complicated? How would `setup.cfg` then look like?

Reply:
In the doc linked (previous comment), there is an example where one can select between `pyproject.toml`, `setup.cfg`, and `setup.py` each showing their own syntax.

Reply:
My guess is it would be something like

```cfg
[options.extras_require]
nvcomp =
    numpy >=1.20
    cupy >=9.5.0
```

Reply:
Make sense and works for me locally :) 
Thanks John!